services/QuillLMS/app/models/teacher_notification.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: teacher_notifications
+#
+#  id                :bigint           not null, primary key
+#  email_sent        :datetime
+#  message_attrs     :jsonb
+#  notification_type :text             not null
+#  created_at        :datetime         not null
+#  updated_at        :datetime         not null
+#  user_id           :bigint           not null
+#
+# Indexes
+#
+#  index_teacher_notifications_on_user_id  (user_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (user_id => users.id)
+#
+class TeacherNotification < ApplicationRecord
+  NOTIFICATION_TYPES = [
```

Comment:
Note, if there is different logic for each of these type, we could consider [STI](https://medium.com/@GeneHFang/single-table-inheritance-in-rails-6-emulating-oop-principles-in-relational-databases-be60c84e0126) here. This is something we could also migrate to later if it's helpful.